agent/lib/kontena/helpers/weave_helper.rb

```diff
@@ -5,11 +5,22 @@ module Kontena
   module Helpers
     module WeaveHelper
 
-      # @return [Boolean]
-      def weave_running?
-        weave = Docker::Container.get('weave') rescue nil
-        return false if weave.nil?
-        weave.info['State']['Running'] == true
+      def network_adapter
+        Celluloid::Actor[:network_adapter]
+      end
+
+      def wait_weave_running?(timeout = 10, &block)
+        wait = Time.now.to_f + timeout
+        until (running = network_adapter.running?) || (wait < Time.now.to_f)
+          yield if block # debugging
+          sleep 0.5
+        end
+        return running
```

Comment:
The return value is ignored... at least log an error if returning with `!running`, if not raising en exception?